cfme/tests/infrastructure/test_kubevirt.py

```diff
@@ -3,23 +3,75 @@
+    pytest.mark.provider([KubeVirtProvider]),
+    pytest.mark.usefixtures('setup_provider')
 ]
 
 
-def test_k6t_provider_crud(provider):
-    """Tests provisioning via PXE
+@pytest.fixture
+def temp_vm(appliance, provider, provisioning):
+    template_name = provisioning['template']
+    vm_name = random_vm_name('k6tvm')
+    prov_data = {'catalog': {'vm_name': vm_name}}
+    vm = appliance.collections.infra_vms.instantiate(name=vm_name,
+                                                     provider=provider,
+                                                     template_name=template_name)
+    note = ('template {} to vm {} on provider {}'.format(template_name, vm_name, provider.key))
+    prov_data.update({
+        'request': {
+            'email': 'template_provisioner@example.com',
+            'first_name': 'Template',
+            'last_name': 'Provisioner',
+            'notes': note}})
+    view = navigate_to(vm, 'Provision')
+    view.form.fill_with(prov_data, on_change=view.form.submit_button)
+    view.flash.assert_no_error()
+    vm.wait_to_appear()
+    yield vm
+    vm.retire()
 
-    Metadata:
-        test_flag: v2v
-    """
-    provider.create()
+
+def test_k6t_provider_crud(provider):
 
     with update(provider):
         provider.name = fauxfactory.gen_alphanumeric() + '_updated'
 
     provider.delete(cancel=False)
     provider.wait_for_delete()
+
+
+@pytest.mark.parametrize('custom_prov_data',
+                         [{}, {'hardware': {'cpu_cores': '8', 'memory': '8192'}}],
+                         ids=['via_lifecycle', 'override_template_values'])
+def test_k6t_vm_crud(request, appliance, provider, provisioning, custom_prov_data):
+    vm_name = random_vm_name('k6tvm')
+    prov_data = {'catalog': {'vm_name': vm_name}}
+    provider.refresh_provider_relationships()
+    prov_data.update(custom_prov_data)
+    template = provisioning['template']
+    do_vm_provisioning(appliance, template, provider, vm_name, prov_data, request,
+                       smtp_test=False, wait=False)
+    logger.info('Waiting for cfme provision request for vm %s', vm_name)
+    request_description = 'Provision from [{}] to [{}]'.format(template, vm_name)
+    provision_request = appliance.collections.requests.instantiate(request_description)
+    provision_request.wait_for_request(method='ui', num_sec=300)
+    assert provision_request.is_succeeded(method='ui'), \
+        ("Provisioning failed with the message {}".format(provision_request.row.last_message.text))
+
+
+@pytest.mark.parametrize('from_details', ['True', 'False'])
+@pytest.mark.parametrize(['power_option', 'vm_state'],
+                         [('Power On', 'currentstate-on'), ('Power Off', 'currentstate-off')])
+def test_vm_power_management(request, appliance, provider, temp_vm,
+                             from_details, power_option, vm_state):
+    temp_vm.power_control_from_cfme(from_details=from_details, option=power_option)
```

Comment:
You need to ensure the VM is in a different state before testing power control - Can't test the power on control if the VM is already on.  Looks like we don't have wrapanapi modules for kubevirt, so please add a TODO

Reply:
Added